lib/adhearsion/call_controller.rb

```diff
@@ -73,11 +73,23 @@ def mixin(mod)
     # @param block to execute on the call
     #
     def initialize(call, metadata = nil, &block)
+      @call_hangup_flag = false
+      if call.alive? && call.active?
+        call.register_event_handler Punchblock::Event::Complete do |event|
+          if event.reason.is_a? Punchblock::Event::Complete::Hangup
+            @call_hangup_flag = true
```

Comment:
Ok, should be done.
Now the problem for me is how to write the test coverage : never done.
Any help or suggestions?